src/pb_stub.cc

```diff
@@ -837,6 +857,92 @@ Stub::GetOrCreateInstance()
   return Stub::stub_instance_;
 }
 
+void
+Stub::LaunchLogRequestThread()
+{
+  log_thread_ = true;
+  log_monitor_ = std::thread(&Stub::ServiceLogRequests, this);
+}
+
+void
+Stub::TerminateLogRequestThread()
+{
+  log_thread_ = false;
+  log_monitor_.join();
+}
+
+void
+Stub::EnqueueLogRequest(std::unique_ptr<PbLog>& log_ptr)
+{
+  std::unique_lock<std::mutex> guard{log_message_mutex_};
+  log_request_buffer.push(std::move(log_ptr));
+}
+
+void
+Stub::ServiceLogRequests()
+{
+  while (log_thread_) {
```

Comment:
This is a busy thread loop which significantly increases the CPU utilization. You need to either use semaphore or a condition variable to avoid busy waiting. If there are no logs, this is a while loop that is always true and just keeps checking the condition.